keys.go

```diff
@@ -302,6 +362,46 @@ type ForceOpt struct {
 	Force bool
 }
 
+//ListKeyVersionsOptions struct to add the query parameters for the ListKeyVersions function
+type ListKeyVersionsOptions struct {
+	Limit      *uint64
+	Offset     *uint64
+	TotalCount *bool
+}
+
+// ListKeyVersions gets all the versions of the key resource by specifying ID of the key and/or optional parameteres
+func (c *Client) ListKeyVersions(ctx context.Context, id string, listKeyVersionsOptions *ListKeyVersionsOptions) (*KeyVersions, int, error) {
+	keyVersion := KeyVersions{}
+	// forming the request
+	req, err := c.newRequest("GET", fmt.Sprintf("keys/%s/versions", id), nil)
+	if err != nil {
+		return nil, 0, err
+	}
+
+	// extracting the query parameters and encoding the same in the request url
+	if listKeyVersionsOptions != nil {
+		values := req.URL.Query()
+		if listKeyVersionsOptions.Limit != nil {
+			values.Set("limit", fmt.Sprint(*listKeyVersionsOptions.Limit))
+		}
+		if listKeyVersionsOptions.Offset != nil {
+			values.Set("offset", fmt.Sprint(*listKeyVersionsOptions.Offset))
+		}
+		if listKeyVersionsOptions.TotalCount != nil {
+			values.Set("totalCount", fmt.Sprint(*listKeyVersionsOptions.TotalCount))
+		}
+		req.URL.RawQuery = values.Encode()
+	}
+
+	//making a request
+	_, err = c.do(ctx, req, &keyVersion)
+	if err != nil {
+		return nil, 0, err
+	}
+	count := keyVersion.Metadata.TotalCount
```

Comment:
What happens here if I specify TotalCount in the ListKeyVersionsOptions as 'false'? Wouldn't this be incorrect or wouldn't the JSON unmarshalling error out because there would be no TotalCount field in the response?

Maybe consider making the count return a `*int` instead of `int`